web_src/js/features/notification.js

```diff
@@ -0,0 +1,94 @@
+const {AppSubUrl, csrf, NotificationSettings} = window.config;
+
+export function initNotificationsTable() {
+  $('#notification_table .button').click(function () {
+    updateNotification(
+      $(this).data('url'),
+      $(this).data('status'),
+      $(this).data('page'),
+      $(this).data('q'),
+      $(this).data('notification-id')
+    ).then((data) => {
+      $('#notification_div').replaceWith(data);
+      initNotificationsTable();
+      updateNotificationCount();
+    });
+    return false;
+  });
+}
```

Comment:
Would refactor to use async/await over `.then` and replace deprecated `.click(fn)` with `.on('click', fn)`:

```js
export function initNotificationsTable() {
  $('#notification_table .button').on('click', async function () {
    const data = await updateNotification(
      $(this).data('url'),
      $(this).data('status'),
      $(this).data('page'),
      $(this).data('q'),
      $(this).data('notification-id')
    );
    
    $('#notification_div').replaceWith(data);
    initNotificationsTable();
    updateNotificationCount();
    return false;
  });
}